src/config.rs

```diff
@@ -0,0 +1,182 @@
+use std::env::VarError;
+use std::error::Error;
+use std::fmt;
+use std::str::FromStr;
+
+const ENVIRONMENT_VARIABLE_PREFIX: &str = "MONITORBOT_";
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    // http server port to bind to
+    pub port: u16,
+    // github api tokens to collect rate limit statistics
+    pub gh_rate_limit_tokens: String,
+    // github rate limit stats data cache refresh rate frequency (in seconds)
+    pub gh_rate_limit_stats_cache_refresh: u64,
+}
+
+impl Config {
+    pub fn from_env() -> Result<Self, ConfigError> {
+        Ok(Self {
+            port: Self::default_env("PORT", 3001)?,
```

Comment:
I would put `default_env` and `require_env` outside of the struct impl, as they don't need to be called outside this module and `Self::` is just noise.

Reply:
done